drivers/ethernet/dsa_nxp_imx_netc.c

```diff
@@ -0,0 +1,297 @@
+ *
+ * @param dev DSA device
+ * @param mac The MAC address to be set in the table
+ * @param fw_port Port number to forward packets
+ * @param tbl_entry_idx The index of entry in the table
+ * @param flags Flags to be set in the entry
+ *
+ * @return 0 if ok, < 0 if error
+ */
+static int dsa_netc_set_mac_table_entry(const struct device *dev, const uint8_t *mac,
+					uint8_t fw_port, uint16_t tbl_entry_idx, uint16_t flags)
+{
+	return 0;
+}
+
+/**
+ * @brief Get DSA MAC address table entry
+ *
+ * @param dev DSA device
+ * @param buf The buffer for data read from the table
+ * @param tbl_entry_idx The index of entry in the table
+ *
+ * @return 0 if ok, < 0 if error
+ */
+static int dsa_netc_get_mac_table_entry(const struct device *dev, uint8_t *buf,
+					uint16_t tbl_entry_idx)
+{
+	return 0;
+}
+
+static void netc_eth_phylink_callback(const struct device *dev, struct phy_link_state *state,
+				      void *user_data)
+{
+	const struct device *ndev = (struct device *)user_data;
+	struct dsa_context *context = ndev->data;
+	struct dsa_netc_data *prv = PRV_DATA(context);
+
+	struct net_if *iface = net_if_lookup_by_dev(ndev);
+	struct ethernet_context *ctx = net_if_l2_data(iface);
+	status_t result;
+
+	if (state->is_up) {
+		LOG_INF("DSA slave port %d Link up", ctx->dsa_port_idx);
+		result = SWT_SetEthPortMII(&prv->swt_handle, ctx->dsa_port_idx,
+					   PHY_TO_NETC_SPEED(state->speed),
+					   PHY_TO_NETC_DUPLEX_MODE(state->speed));
+		if (result != kStatus_Success) {
+			LOG_ERR("DSA slave port %d failed to set MAC up", ctx->dsa_port_idx);
+		}
+		net_eth_carrier_on(iface);
+	} else {
+		LOG_INF("DSA slave port %d Link down", ctx->dsa_port_idx);
+		net_eth_carrier_off(iface);
+	}
+}
+
+static void dsa_netc_iface_init(struct net_if *iface)
+{
+	const struct device *dev = net_if_get_device(iface);
+	struct dsa_netc_slave_config *cfg = (struct dsa_netc_slave_config *)dev->config;
+	struct dsa_context *context = dev->data;
+	struct dsa_netc_data *prv = PRV_DATA(context);
+
+	struct ethernet_context *ctx = net_if_l2_data(iface);
+	struct ethernet_context *ctx_master;
+	int i = cfg->port_idx;
+
+	/* Find master port */
+	if (context->iface_master == NULL) {
+		context->iface_master = net_if_lookup_by_dev(prv->dev_master);
+		if (context->iface_master == NULL) {
+			LOG_ERR("DSA: Master iface NOT found!");
+			return;
+		}
+
+		/*
+		 * Provide pointer to DSA context to master's eth interface
+		 * struct ethernet_context
+		 */
+		ctx_master = net_if_l2_data(context->iface_master);
+		ctx_master->dsa_ctx = context;
+	}
+
+	if (context->iface_slave[i] == NULL) {
+		context->iface_slave[i] = iface;
+		net_if_set_link_addr(iface, cfg->mac_addr, sizeof(cfg->mac_addr),
+				     NET_LINK_ETHERNET);
+		ctx->dsa_port_idx = i;
+		ctx->dsa_ctx = context;
+
+		/*
+		 * Initialize ethernet context 'work' for this iface to
+		 * be able to monitor the carrier status.
+		 */
+		ethernet_init(iface);
+	}
+
+	prv->iface_init_count++;
```

Comment:
i'm not seeing where is this value used, is it needed ?

Reply:
Will remove it. Thanks.